lib/document/Document.js

```diff
@@ -241,6 +268,32 @@ class Document {
     return json;
   }
 
+  /**
+   * Return Document as plain object (without converting encoded fields)
+   *
+   * @return {RawDocument}
+   */
+  toObject() {
+    const plainObject = {
```

Comment:
I would reuse this logic in both methods. We can add a private method like getSystemProperties() and then add either encoded in `toJSON` or "as is" data in `toObject`